### DISCUSSION TITLE
How to compare PNM with DNS?

### DISCUSSION THREAD

**Discussion by pengzhulin** [PMEAL/OpenPNM#2940] [Q&A]
Dear Prof. Gostick,
Thank you for your extraordinary contributions to the field. I am interested in learning how to compare PNM with DNS simulations, importing an extracted network and predicting transport properties (https://openpnm.org/examples/applications/network_extraction.html).

Would you mind sharing the DNS code or any related tutorials? I would like to compare DNS and PNM using our own geometry matrix.

I greatly appreciate your valuable insights and help!

Thank you very much!

**Comment by jgostick** [comment]
For dns we currently use the lbm package from the UT Austin group for permeability and I recommend taufactor2 for diffusivity.  We also have the drainage function in porespy if you have hg intrusion data.

**Comment by pengzhulin** [comment]
Dear Prof. Gostick,
Thank you very much for your reply and comments. 

(1) Using drainage and hg instrusion data is really a good idea.
(2) Your comments and replys are highly appreciated!

However, I am still curious about the DNS part. I guess that it must be MPLBM-ut (https://github.com/je-santos/MPLBM-UT). And I try to use this LBM to study DNS mentioned in OpenPNM (https://openpnm.org/examples/applications/network_extraction.html). Would you mind having a quick look about my proceudure?
(1) read the '../../tests/fixtures/berea_100_to_300.npz' matrix
(2) transfrom it to .raw as input in MPLBM-ut
(3) use MPLBM-ut to calculate the permeability, K is about 11 lbm_unit^2. So, K_LBM = 11* (5.345e-6 m)^2 = 314.3e-12 m^2.
(4) K_LBM this result is far from Kx = 1360 mDarcy ~ 1.36 e-12 m^2.

Do you think the procedure is correct? Thank you very very much for your time and your kind help!

**Reply by jgostick** [reply]
The procedure looks ok. We usually get good results from that lbm package, so maybe you should double check your procedure on a different image?  Get something from the digital rocks portal which has a known K, and make sure you get the same as them.  In fact, Javier Santos has pushed a huge batch of results there which he used his MPLBM-ut code on, so make sure you're matching that.  It's possible that your conversion to .raw is not working as you think, for instance.
